src/buildvars.py

```diff
@@ -119,6 +120,37 @@ def _update_remote_bvars(stackname, buildvars):
     put(StringIO(encoded), "/etc/build-vars.json.b64", use_sudo=True)
     LOG.info("%r updated", stackname)
 
+#
+#
+#
 
-def _bvarstype(bvars):
-    return FULL
+@requires_aws_stack
+def refresh(stackname, context):
+    "(safely) replaces the buildvars file on the ec2 instance(s)"
+
+    def _refresh_buildvars():
+        old_buildvars = _retrieve_build_vars()
+
+        node = old_buildvars.get('node')
+        if not node or not str(node).isdigit():
```

Comment:
I did something stupid here and `node` was set to `None` at some point so a lot of the code below is just to handle bad node values that probably won't ever exist again.